### ISSUE TITLE
Configurable columns for listview

### ISSUE THREAD

**Issue by jelly** [cockpit-project/cockpit-files#564]
We would like to add two more columns to the listview:

* permission - showing `-rw-r--r--`. Do we sort this column based on the octal value so 777 > 600?
* user/group - likely a separate column for both? (Sorting of course uses `localCompare`

We likely don't want to add these columns by default but allow a user to select what they want to show for which we have no UI.

**Comment by garrett** [comment]
So I think we probably want to remove the sidebar (I'm working on that; we can have a status bar with the critical information instead) and we'd have enough space for owner (special merge of user+group) and permission.

Here's Nautilus has for customized columns in details mode:

![image](https://github.com/cockpit-project/cockpit-files/assets/10246/142a228f-ed33-4866-9b36-370562fecbb6)

But, again: Nautilus is a desktop file manager and we're on a server, etc., etc.

Nautilus doesn't have these most of these items on by default. 

On a multi-user system, I think we might want different defaults. Permissions and ownership are more important on a multi-user server system than on a (usually, mostly) single-user desktop system.

**Comment by garrett** [comment]
I think we need to split out the parts of details that we want to have by default first, before even considering customization.

That is, we should have a PR for permissions first, then consider what to do about ownership, then consider if we even want customization.

**Comment by garrett** [comment]
For showing permissions, I think the only thing that makes sense is to have the descriptive form with triplet attributes (not octal), and have descriptions for it in a tooltip to make it more approachable, like this:

![permissions in details, with a tooltip](https://github.com/cockpit-project/cockpit-files/assets/10246/67bdf803-ce6b-4d2b-86f9-dba7acac10f7)

Note the spacing, lack of `d` (as we have another place to show if it's a directory or file`), and grouping. The tooltip has alignment on the labels and for the content, so the content aligns (and isn't ragged due to being all in one line).

**Comment by allisonkarlitskaya** [comment]
Please make sure to support `s` (`+xs`), `S` (`+s-x`) and `t` (sticky) in the output for permissions.

**Comment by garrett** [comment]
I don't think we want to add configurability, at least not until we need to. Here are all the fields for details that I think we might want to include. It's not too much:

![Board(3)](https://github.com/cockpit-project/cockpit-files/assets/10246/d3cc83ea-db5d-4f81-94b2-5f40ac21d2e3)

I don't think we should show octals. People generally don't understand that as well as the symbolic variants. If we split it into triplets like in the mockup, then it becomes even more obvious which is user, group, and others.

Note, _not related to this PR, but related to the mockup_: I collapsed `user:group` into owner, as most distributions use "User Private Groups" (UPG), where user:group is basically always user:user by default, so showing just "user" would be the main default, and you can easily then see when something is assigned to another group separate from the user.
